### PR TITLE
Code Quality: Upgrade `.gitignore` to latest

### PR DESCRIPTION
**Pull request by Lamparter**
Removed a few ignores such as the '.nupkg' ignore since this would conflict with the NuGet package in the 'lib' directory

----

### Before the change?

* `.gitignore` was the visual studio ignore at the time of project initialisation 

### After the change?

* Now the latest from [the gitignore repo](https://github.com/github/gitignore)

This is a pretty minor code quality change but it's good to update files like these to future-proof the repo and prevent any accidents.

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No

----

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,15 @@
 ## Ignore Visual Studio temporary files, build results, and
 ## files generated by popular Visual Studio add-ons.
+##
+## Get latest from https://github.com/github/gitignore/blob/main/VisualStudio.gitignore
+
+# Custom files
```
Comment (0x5bfa) [inline .gitignore:6]: I think these are no longer needed

**Review by nickfloyd — APPROVED** [review]

**Comment by nickfloyd** [comment]
Thanks for this change... would you also mind updating `actions/upload-artifact` to v3 or v4 in [dotnetcore.yaml](https://github.com/octokit/octokit.graphql.net/blob/c8d9028205474ae8547daf8332827f4b028a6809/.github/workflows/dotnetcore.yml#L48) to get CI ✅

File: .gitignore (hunk 1 above)
Comment (nickfloyd) [reply .gitignore:6]: Correct, those are legacy and can be removed.

Comment (Lamparter) [reply .gitignore:6]: Yeah, was going to fix this this morning on my phone but the internet cut out 😅

**Comment by 0x5bfa** [comment]
> would you also mind updating actions/upload-artifact to v3 or v4 in [dotnetcore.yaml](https://github.com/octokit/octokit.graphql.net/blob/c8d9028205474ae8547daf8332827f4b028a6809/.github/workflows/dotnetcore.yml#L48) to get CI ✅

I did this in other pr. The pr that contains the move to src dir.

**Comment by Lamparter** [comment]
Yeah I think @nickfloyd just means he wants to see the build result here 😅

**Review by nickfloyd — APPROVED** [review]

**Pull request merged**